pkg/didmethod/httpuniversal/resolver.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package httpuniversal
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	errors "golang.org/x/xerrors"
+)
+
+//HTTPClient abstracts away http client library
+type HTTPClient interface {
+	Get(url string) (body []byte, err error)
+}
+
+//StandardHTTPClient is based on http.Client
+type StandardHTTPClient struct {
+	client *http.Client
+}
+
+//Get implements HTTP Get operation on standard http client
+func (shc StandardHTTPClient) Get(url string) (body []byte, err error) {
+	resp, err := shc.client.Get(url)
+	if err != nil {
+		return nil, errors.Errorf("HTTP Get request failed: %w", err)
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Errorf("Failed to read response body: %w", err)
+	}
+
+	err = resp.Body.Close()
+	if err != nil {
+		return nil, errors.Errorf("Failed to close response body: %w", err)
+	}
+	return b, nil
+}
+
+// URLBuilderCreator is a factory of URL builder based on base url and did
+type URLBuilderCreator func(baseURL, did string) (string, error)
+
+// QueryParamURLBuilder provides a factory of URL builder using query parameters
+func QueryParamURLBuilder(paramName string) URLBuilderCreator {
```

Comment:
I don't think we need this additional level of indirection now, do we? I don't see definitions for different strategies for building URLs

Reply:
Agree. This is simplified.